src/lib.rs

```diff
@@ -145,3 +158,52 @@ impl DeviceManager {
         Ok(())
     }
 }
+
+async fn get_device_id(opt_device_id: Option<String>) -> Result<String, DeviceManagerError> {
+    return if opt_device_id != None {
+        Ok(opt_device_id.unwrap())
```

Comment:
Is this a manual version of `if let`?
Also, you can delete the `return if` and remove the final ;

Reply:
It is also available the `is_none` function :grin: https://doc.rust-lang.org/std/option/enum.Option.html#method.is_none

Reply:
Sure, but I only write because `option != None` was horrible to see. :grinning: